eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -0,0 +1,119 @@
+//! Implementation of YASHE cryptosystem
+//! `<https://eprint.iacr.org/2013/075.pdf>`
+
+use crate::primitives::poly::{inverse, modular_poly::Poly, Coeff};
+use ark_ff::{One, UniformRand, Zero};
+use rand::rngs::ThreadRng;
+use rand_distr::{Distribution, Normal};
+
+pub mod test;
+
+/// Yashe parameters
+#[derive(Clone)]
+pub struct YasheParams {
+    /// Plaintext coefficient modulus
+    pub t: u64,
+    /// Standard deviation
+    pub delta: f64,
+}
+
+/// Yashe scheme
+pub struct Yashe<const MAX_POLY_DEGREE: usize> {
+    /// Cryptosystem parameters
+    params: YasheParams,
+}
+
+/// Private key struct
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct PrivateKey<const MAX_POLY_DEGREE: usize> {
+    /// Sampled with small coefficients (and invertible)
+    f: Poly<MAX_POLY_DEGREE>,
+    /// The inverse of f
+    finv: Poly<MAX_POLY_DEGREE>,
+    /// Private key
+    priv_key: Poly<MAX_POLY_DEGREE>,
+}
+
+/// Public key struct
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct PublicKey<const MAX_POLY_DEGREE: usize> {
+    /// Public key
+    h: Poly<MAX_POLY_DEGREE>,
+}
```

Comment:
Are these `allow(dead_code)` still needed?

Usually they indicate that the type is unused _and_ not exported publicly. But these types should be public, if they're not that's a bug.

Or maybe there are some fields which are never read or written? That also seems like it could be a bug. Maybe the field could be deleted, or it needs a public accessor function?

```suggestion
/// Private key struct
#[derive(Debug, Clone)]
pub struct PrivateKey<const MAX_POLY_DEGREE: usize> {
    /// Sampled with small coefficients (and invertible)
    f: Poly<MAX_POLY_DEGREE>,
    /// The inverse of f
    finv: Poly<MAX_POLY_DEGREE>,
    /// Private key
    priv_key: Poly<MAX_POLY_DEGREE>,
}

/// Public key struct
#[derive(Debug)]
pub struct PublicKey<const MAX_POLY_DEGREE: usize> {
    /// Public key
    h: Poly<MAX_POLY_DEGREE>,
}
```